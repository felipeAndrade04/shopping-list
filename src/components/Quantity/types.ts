/* eslint-disable no-unused-vars */
export interface QuantityProps {
  initialValue?: number;
  value?: number;
  onChange?: (value: number) => void;
}
