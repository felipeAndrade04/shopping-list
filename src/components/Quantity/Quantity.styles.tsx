import styled from 'styled-components/native';

export const QuantityComponent = styled.View`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const QuantityButton = styled.TouchableOpacity`
  padding: 2px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: ${({ theme }) => theme.colors.main};
  border-radius: 4px;
`;

export const QuantityValue = styled.Text`
  padding: 0px 8px;
  font-size: 16px;
  font-weight: ${({ theme }) => theme.fonts.medium};
`;
