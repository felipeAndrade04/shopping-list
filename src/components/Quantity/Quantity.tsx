import { useState } from 'react';
import { AntDesign } from '@expo/vector-icons';
import * as S from './Quantity.styles';
import { QuantityProps } from './types';

export function Quantity({ initialValue, value, onChange }: QuantityProps) {
  const [quantity, setQuantity] = useState(initialValue || value || 1);

  function onIncrement() {
    const newValue = quantity + 1;

    onChange && onChange(newValue);
    setQuantity(newValue);
  }

  function onDecrement() {
    const newValue = quantity - 1;

    onChange && onChange(newValue);
    setQuantity(newValue);
  }

  return (
    <S.QuantityComponent testID="quantity">
      <S.QuantityButton onPress={onDecrement} testID="decrement-quantity">
        <AntDesign name="minus" size={18} color="white" />
      </S.QuantityButton>
      <S.QuantityValue>{quantity}</S.QuantityValue>
      <S.QuantityButton onPress={onIncrement} testID="increment-quantity">
        <AntDesign name="plus" size={18} color="white" />
      </S.QuantityButton>
    </S.QuantityComponent>
  );
}
