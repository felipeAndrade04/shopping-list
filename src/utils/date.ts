const months = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];

export function formatDate(date: Date) {
  const day = date.getDay();
  const month = date.getMonth();

  return `${day} de ${months[month]}`;
}
