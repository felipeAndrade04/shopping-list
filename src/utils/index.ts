export * from './delay';
export * from './date';
