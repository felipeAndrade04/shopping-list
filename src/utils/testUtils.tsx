import React from 'react';
import { authReducer } from '@app/store';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import * as theme from '@app/theme';
import { configureStore } from '@reduxjs/toolkit';

let store = configureStore({
  reducer: {
    auth: authReducer,
  },
});

export const wrapper = ({ children }: { children: ReactNode }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>{children}</Provider>
  </ThemeProvider>
);

export function resetStore() {
  store = configureStore({
    reducer: {
      auth: authReducer,
    },
  });
}
