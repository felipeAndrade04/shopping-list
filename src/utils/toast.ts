import EventManager from './EvenManager';

export type ToastType = 'success' | 'info' | 'error';

export interface Toast {
  id?: string;
  type: ToastType;
  text: string;
  duration?: number;
}

export const toastEventManager = new EventManager();

export default function toast({ type, text, duration }: Toast) {
  toastEventManager.emit<Toast>('addtoast', {
    id: new Date().getTime().toString(),
    type,
    text,
    duration,
  });
}
