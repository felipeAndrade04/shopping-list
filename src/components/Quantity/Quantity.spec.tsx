import { fireEvent, render } from '@testing-library/react-native';
import { Quantity } from './Quantity';
import { wrapper } from '@app/utils/testUtils';

describe('<Quantity />', () => {
  it('Should be render correctly', () => {
    const tree = render(<Quantity />, { wrapper }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should show default quantity', () => {
    const { getByText } = render(<Quantity />, { wrapper });

    expect(getByText('1')).toBeTruthy();
  });

  it('Should show initial quantity', () => {
    const { getByText } = render(<Quantity initialValue={2} />, { wrapper });

    expect(getByText('2')).toBeTruthy();
  });

  it('Should increment quantity', () => {
    let quantity = 1;
    const { getByText, getByTestId } = render(
      <Quantity onChange={(newQuantity) => (quantity = newQuantity)} initialValue={quantity} />,
      { wrapper }
    );

    expect(getByText('1')).toBeTruthy();
    expect(quantity).toEqual(1);

    const button = getByTestId('increment-quantity');
    fireEvent.press(button);

    expect(getByText('2')).toBeTruthy();
    expect(quantity).toEqual(2);
  });

  it('Should decrement quantity', () => {
    let quantity = 2;
    const { getByText, getByTestId } = render(
      <Quantity onChange={(newQuantity) => (quantity = newQuantity)} initialValue={quantity} />,
      { wrapper }
    );

    expect(getByText('2')).toBeTruthy();
    expect(quantity).toEqual(2);

    const button = getByTestId('decrement-quantity');
    fireEvent.press(button);

    expect(getByText('1')).toBeTruthy();
    expect(quantity).toEqual(1);
  });
});
