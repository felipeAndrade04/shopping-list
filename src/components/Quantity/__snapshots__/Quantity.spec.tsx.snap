// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quantity /> Should be render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      },
    ]
  }
  testID="quantity"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#009Cf7",
        "borderRadius": 4,
        "display": "flex",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 2,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 2,
      }
    }
    testID="decrement-quantity"
  >
    <Text />
  </View>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontWeight": "Poppins-Medium",
          "paddingBottom": 0,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 0,
        },
      ]
    }
  >
    1
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#009Cf7",
        "borderRadius": 4,
        "display": "flex",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 2,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 2,
      }
    }
    testID="increment-quantity"
  >
    <Text />
  </View>
</View>
`;
